tests/integration/conftest.py

```diff
@@ -370,3 +421,73 @@ def is_mysql_ready():
         ),
     )
     await ops_test.model.wait_for_idle()
+
+
+@pytest.fixture(scope="module", name="test_image")
+def image_fixture():
+    """A PNG image that can be used in tests."""
+    image_response = requests.get(
+        "https://s.w.org/style/images/about/WordPress-logotype-wmark.png", timeout=10
+    )
+    assert image_response.status_code == 200
+    return image_response.content
+
+
+@pytest.fixture(scope="module", name="swift_conn")
+def swift_conn_fixture(openstack_environment):
+    """Create a swift connection client."""
+    if openstack_environment is None:
+        return None
```

Comment:
Are we ok with returning `None`? If a test depends on this fixture but receives `None`, would you expect the test to still work?

Reply:
No, the test should fail as intended.